mllib/src/main/scala/org/apache/spark/ml/regression/LinearRegression.scala

```diff
@@ -42,34 +50,122 @@ private[regression] trait LinearRegressionParams extends RegressorParams
 class LinearRegression extends Regressor[Vector, LinearRegression, LinearRegressionModel]
   with LinearRegressionParams {
 
-  setDefault(regParam -> 0.1, maxIter -> 100)
-
-  /** @group setParam */
+  /**
+   * Set the regularization parameter.
+   * Default is 0.0.
+   * @group setParam
+   */
   def setRegParam(value: Double): this.type = set(regParam, value)
+  setDefault(regParam -> 0.0)
```

Comment:
To match R's default result, we need to `0.0`. Also, the meaning of lambda will be changed if the numbers of sample is changed. So it's hard to come out with a good default. Why don't we implement regularization path to find the best lambda?